exporter/datadogexporter/traces_exporter.go

```diff
@@ -187,7 +188,7 @@ func newTraceAgent(ctx context.Context, params exporter.Settings, cfg *Config, s
 	if err != nil {
 		return nil, err
 	}
-	return agent.NewAgent(ctx, acfg, telemetry.NewNoopCollector(), metricsClient), nil
+	return agent.NewAgent(ctx, acfg, telemetry.NewNoopCollector(), metricsClient, gzip.NewComponent()), nil
```

Comment:
This is backwards-compatible, see https://github.com/DataDog/datadog-agent/pull/26775/files#diff-7c1162a18d33140731993a632d7a80ecda3c1b7e493a4d5c454cf6794c3bc29bR287